File: 199_binary_tree_right_side_view/rightSideView.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    public:
        vector<int> rightSideView(TreeNode* root) {
            vector<int> res;
    
            dfs(root, 0, res);
    
            return res;
        }
    
    private:
        void dfs(TreeNode* root, int level, vector<int>& res) {
            if (root) {
                if (res.size() <= level) res.push_back(root->val);
    
                res[level] = root->val;
    
                dfs(root->left, level + 1, res);
                dfs(root->right, level + 1, res);
            }
        }
    };

File: 199_binary_tree_right_side_view/rightSideView.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []

        self.dfs(root, 0, res)

        return res
        
    def dfs(self, root: Optional[TreeNode], level: int, res: List[int]):
        if root:
            if len(res) <= level:
                res.append(root.val)
            res[level] = root.val
            self.dfs(root.left, level + 1, res)
            self.dfs(root.right, level + 1, res)
                
        return